src/lib.rs

```diff
@@ -86,6 +86,30 @@ fn unsafe_alloc_buffer(len: usize) -> Vec<u8> {
     ret
 }
 
+#[derive(Default, Copy, Clone)]
+pub struct Uint128 {
+    pub bytes: [u8; 16],
+}
+
+#[derive(Default, Copy, Clone)]
+pub struct Bytes20 {
+    pub bytes: [u8; 20],
+}
+
+#[derive(Default, Copy, Clone)]
+pub struct Bytes32 {
+    pub bytes: [u8; 32],
+}
+
+type EtherValue = Uint128;
+type Address = Bytes20;
+type StorageKey = Bytes32;
+type StorageValue = Bytes32;
+type Topic = Bytes32;
+type Hash = Bytes32;
+// TODO: is this a number of a bytestring?
```

Comment:
this is a number. https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2.md#specification